### PR TITLE
configured pep8 and pytest

### PR DESCRIPTION
**Pull request by mpoegel**
Implements #5 

* `npm run lint-py` to lint the python code
* changed the JS lint command to `npm run lint-js`
* `npm test` also runs `pytest`
* updated travis-ci

### CODE REVIEW

**Comment by droghio** [comment]
Looks like the requirements.txt broke the build system. @mpoegel can you confirm whether this is a singleton oddity or an actual break caused by the commit?

**Comment by mpoegel** [comment]
I still haven't figured out what went wrong in the build process. The requirements.txt file works; I think there's just a problem installing python or pip.

**Comment by mpoegel** [comment]
Fixed the issue with [this](https://github.com/travis-ci/travis-ci/issues/1705)

File: transpile/transpiler.py [hunk 1]
```diff
@@ -9,53 +9,60 @@
 # json.load
 import json
 
+
 class Transpiler():
-    '''The class that handles file loading and compilation/uploading'''
+    """he class that handles file loading and compilation/uploading"""
```
Comment (droghio) [inline transpile/transpiler.py]: So close, yet so far.

File: transpile/transpiler.py [hunk 2]
```diff
@@ -9,53 +9,60 @@
...
         self.__load()
         self.__convert()
         self.__compile()
-        #self.__upload()
+        # self.__upload()
         # TODO return an error if any are encountered
-    
+
     def __upload(self):
-        '''Uploads the binary file to the Arduino on the given port with avrdude'''
```
Comment (droghio) [inline transpile/transpiler.py]: Good of you to pick on U too.

File: transpile/transpiler.py [hunk 3]
```diff
@@ -9,53 +9,60 @@
...
+        """ploads the binary file to the Arduino on the given port with avrdude"""
         # Execute avrdude to upload the Intel hex file to the Arduino
-        subprocess.call('avrdude -p m328p -P {0} -c arduino -b 19200 -F -u -U flash:w:{1}'.format(self._port, self._binaryFile))
-    
+        subprocess.call('avrdude -p m328p -P {0} -c arduino -b 19200 -F -u -U flash:w:{1}'
+                        .format(self._port, self._binaryFile))
+
     def __compile(self):
-        '''Compiles the given Arduino C file to a binary file using avr-g++'''
+        """ompiles the given Arduino C file to a binary file using avr-g++"""
```
Comment (droghio) [inline transpile/transpiler.py]: I C you have problems here.

File: transpile/transpiler.py [hunk 4]
```diff
@@ -9,53 +9,60 @@
...
         subprocess.call('avr-objcopy -R .eeprom -O ihex temp.o {0}'.format(self._binaryFile))
         # Remove the temporary object file
         try:
             os.remove("temp.o")
         except OSError:
             pass
-    
+
     def __convert(self):
-        '''Converts the user's block diagram to C code using the appropriate Converter'''
```
Comment (droghio) [inline transpile/transpiler.py]: I C, I c.

File: transpile/transpiler.py [hunk 5]
```diff
@@ -9,53 +9,60 @@
...
+        """onverts the user's block diagram to C code using the appropriate Converter"""
         self._converter.convert(self._rootBlock)
-    
+
     def __load(self):
-        '''Loads the input file and saves the root block'''
+        """oads the input file and saves the root block"""
```
Comment (droghio) [inline transpile/transpiler.py]: Lo_

File: transpile/transpiler.py [hunk 6]
```diff
@@ -71,20 +78,22 @@ def createBlock(blockDictionary, index):
...
     for index in range(0, len(blockDictionary)):
         curDict = blockDictionary[index]
         curBlock = createBlock(curDict, index)
         blockArray[index] = curBlock
     return blockArray
 
+
 def populateAttributes(blockDictionary, blocks):
-    '''Populates the attributes of the blocks from the dictionary'''
+    """opulates the attributes of the blocks from the dictionary"""
```
Comment (droghio) [inline transpile/transpiler.py]: _oobar

**Pull request merged**